GuessTheFlag/Controller/ResultBoardViewController.swift

```diff
@@ -78,10 +81,30 @@ class ResultBoardViewController: UIViewController {
             
             saveGameButton.leadingAnchor.constraint(equalTo: conditionalMessage.leadingAnchor),
             saveGameButton.trailingAnchor.constraint(equalTo: conditionalMessage.trailingAnchor),
-            saveGameButton.bottomAnchor.constraint(equalTo: scoreBackgroundView.bottomAnchor, constant: -40)
+            saveGameButton.bottomAnchor.constraint(equalTo: scoreBackgroundView.bottomAnchor, constant: -20)
         ])
     }
     
+    private func handleConditionalMessage() -> String {
```

Comment:
Para este caso não vejo muito problema, mas apenas para conhecimento: uma forma de organizar a view controller de forma que ela não tenha muitas responsabilidades seria extrair essas lógicas para outro lugar (para o arquivo de view model, por exemplo). Assim a view controller ficaria responsável apenas por montar e apresentar a view, e tratar as interações.

Isto facilita também caso você queira implementar testes unitários. A lógica estando separada da view, você somente testaria a lógica. A view não precisaria de testes pois ela seria um produto do resultado da lógica, ou seja, apenas mostraria os componentes e os valores para o usuário.

Reply:
Aqui eu até concordo contigo mas eu penso em deixar dessa forma, pelo menos por enquanto. Realmente eu já to sentindo que ta ficando muito grande a classe e começa a complicar a compreensão de cada uma das funções.